core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -137,7 +138,13 @@ private[spark] class EventLoggingListener(
     // scalastyle:on println
     if (flushLogger) {
       writer.foreach(_.flush())
-      hadoopDataStream.foreach(_.hflush())
+      hadoopDataStream.foreach(ds => {
```

Comment:
OK, and it's not better to just call hsync in all cases -- you have to special case this?